internal/util/update/update_test.go

```diff
@@ -1499,6 +1541,446 @@ func TestCommand_Run_subpackages(t *testing.T) {
 				},
 			},
 		},
+		{
+			name: "updated setters on root package",
+			initialUpstream: pkgbuilder.NewRootPkg().
+				WithKptfile(
+					pkgbuilder.NewKptfile().
+						WithSetters(
+							pkgbuilder.NewSetter("my-setter", "foo"),
+						),
+				),
+			updatedUpstream: testutil.Content{
+				Pkg: pkgbuilder.NewRootPkg().
+					WithKptfile(
+						pkgbuilder.NewKptfile().
+							WithSetters(
+								pkgbuilder.NewSetter("my-setter", "bar"),
+							),
+					),
+			},
+			expectedResults: []resultForStrategy{
+				{
+					strategies: []StrategyType{FastForward, KResourceMerge},
+					expectedLocal: pkgbuilder.NewRootPkg().
+						WithKptfile(
+							pkgbuilder.NewKptfile().
+								WithUpstreamRef("upstream", "/", "master").
+								WithSetters(
+									pkgbuilder.NewSetter("my-setter", "bar"),
+								),
+						),
+				},
+			},
+		},
+		{
+			name: "with remote subpackages",
```

Comment:
This is test is tough to understand as to what it is doing. Can you add some comments.

Reply:
I updated the name of the test. The tests are a bit hard to read at first, but I think it is pretty readable considering it involves some complex setup.